### PR TITLE
Replace the PatchOperation with the PatchSet

### PR DESCRIPTION
**Pull request by alicefr**
### What this PR does
Before this PR:
Directly using the [PatchOperation struct](https://github.com/kubevirt/kubevirt/blob/main/pkg/apimachinery/patch/patch.go#L28)

After this PR:
We are trying to simplify the patch operations and use everywhere in the code the [PatchSet](https://github.com/kubevirt/kubevirt/blob/main/pkg/apimachinery/patch/patch.go#L68). Eventually, we could set the PatchOperation struct as private.

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/virt-api/webhooks/mutating-webhook/mutators/migration-create-mutator_test.go [hunk 1]
```diff
@@ -102,11 +102,5 @@ func expectedMigrationObjectMeta(currentObjectMeta k8smetav1.ObjectMeta, vmiName
...
 
 func expectedJSONPatchForVMIMCreation(expectedObjectMeta k8smetav1.ObjectMeta) ([]byte, error) {
-	return patch.GeneratePatchPayload(
-		patch.PatchOperation{
-			Op:    patch.PatchReplaceOp,
-			Path:  "/metadata",
-			Value: expectedObjectMeta,
-		},
-	)
+	return patch.New(patch.WithReplace("/metadata", expectedObjectMeta)).GeneratePayload()
```
Comment (fossedihelm) [inline pkg/virt-api/webhooks/mutating-webhook/mutators/migration-create-mutator_test.go:105]: we can entirely get rid of this function, performing patch from the caller

**Review by fossedihelm — APPROVED** [review]
Looks good! Thanks

File: pkg/virt-api/webhooks/mutating-webhook/mutators/migration-create-mutator_test.go (hunk 1 above)
Comment (alicefr) [reply pkg/virt-api/webhooks/mutating-webhook/mutators/migration-create-mutator_test.go]: Done

**Review by fossedihelm — APPROVED** [review]

**Comment by alicefr** [comment]
@xpivarc can you please take a look when you have a bit of time

File: pkg/virt-controller/watch/node.go [hunk 2]
```diff
@@ -274,7 +274,13 @@ func (c *NodeController) createAndApplyFailedVMINodeUnresponsivePatch(vmi *virtv
 	c.recorder.Event(vmi, v1.EventTypeNormal, NodeUnresponsiveReason, fmt.Sprintf("virt-handler on node %s is not responsive, marking VMI as failed", vmi.Status.NodeName))
 	logger.V(2).Infof("Moving vmi %s in namespace %s on unresponsive node to failed state", vmi.Name, vmi.Namespace)
 
-	patchBytes, err := generateFailedVMIPatch(vmi.Status.Reason)
+	patchSet := patch.New(patch.WithReplace("/status/phase", virtv1.Failed))
+	if vmi.Status.Reason == "" {
+		patchSet.AddOption(patch.WithAdd("/status/reason", NodeUnresponsiveReason))
+	} else {
+		patchSet.AddOption(patch.WithReplace("/status/reason", NodeUnresponsiveReason))
+	}
```
Comment (xpivarc) [inline pkg/virt-controller/watch/node.go:282]: Isn't add enough?

**Review by xpivarc — COMMENTED** [review]
/approve
/hold
Holding to give time for answering the one question

File: pkg/virt-controller/watch/node.go (hunk 2 above)
Comment (alicefr) [reply pkg/virt-controller/watch/node.go:282]: @xpivarc I think so. The new code is replicated what the old one was doing, but let me change it

**Review by fossedihelm — APPROVED** [review]
Thank you!
@xpivarc feel free to unhold if the comment is addressed

File: pkg/virt-controller/watch/node.go (hunk 2 above)
Comment (alicefr) [reply pkg/virt-controller/watch/node.go]: Done

**Comment by alicefr** [comment]
@xpivarc can we unhold this?

**Pull request merged**